src/util.cpp

```diff
@@ -798,8 +788,9 @@ void AllocateFileRange(FILE *file, unsigned int offset, unsigned int length) {
         fcntl(fileno(file), F_PREALLOCATE, &fst);
     }
     ftruncate(fileno(file), fst.fst_length);
-#elif defined(__linux__)
-    // Version using posix_fallocate
+#elif _POSIX_C_SOURCE >= 200112L
```

Comment:
I think @laanwj's comment about _POSIX_C_SOURCE at https://github.com/bitcoin/bitcoin/pull/12495#discussion_r169970767 applies here as well, so this check is not right.